pkg/liqoctl/status/pods_test.go

```diff
@@ -0,0 +1,438 @@
+			Context("getting images", func() {
+				It("should return a slice of length 1", func() {
+					im := podS.getImages()
+					Expect(len(im)).To(BeNumerically("==", 1))
+					Expect(im[0]).To(Equal(image))
+				})
+			})
+
+			Context("setting images", func() {
+				It("should set the new images", func() {
+					newImages := []string{"image1", "image2"}
+					podS.setImages(newImages)
+					Expect(len(podS.imageVersions)).To(BeNumerically("==", 2))
+					Expect(podS.imageVersions).To(Equal(newImages))
+				})
+			})
+
+			Context("adding an image to the existing ones", func() {
+				When("the image does not exist", func() {
+					It("should append the image to the existing ones", func() {
+						newImage := "newImage"
+						podS.addImageVersion(newImage)
+						Expect(len(podS.imageVersions)).To(BeNumerically("==", 2))
+						Expect(podS.imageVersions[1]).To(Equal(newImage))
+					})
+				})
+
+				When("the image exists", func() {
+					It("should not append the image", func() {
+						podS.addImageVersion(image)
+						Expect(len(podS.imageVersions)).To(BeNumerically("==", 1))
+					})
+				})
+			})
+		})
+
+		Describe("handling errorsPerPod field", func() {
+			var (
+				podName = "pod1"
+				err     = errors.New("new pod error")
+			)
+
+			Context("adding error per pod", func() {
+				When("no errors per pod exists", func() {
+					It("should add the error for the given pod", func() {
+						podS.errors = nil
+						podS.addErrorPerPod(podName, err)
+						Expect(len(podS.errors[podName].errors)).To(BeNumerically("==", 1))
+						Expect(podS.errors[podName].errors[0]).To(MatchError(err))
+					})
+				})
+
+				When("errors per pod exists", func() {
+					It("should append the new error for the given pod", func() {
+						podS.addErrorPerPod(podName, err)
+						Expect(len(podS.errors[podName].errors)).To(BeNumerically("==", 2))
+						Expect(podS.errors[podName].errors[1]).To(MatchError(err))
+					})
+				})
+			})
+		})
+
+		Describe("formatting the podState", func() {
+			When("there are no unavailable pods", func() {
+				It("string should not contain unavailable pods", func() {
+					Expect(podS.format()).NotTo(ContainSubstring("Unavailable"))
+				})
+
+			})
+
+			When("not all pods are ready", func() {
+				It("output should contain unavailable pods", func() {
+					podS.desired = 2
+					podS.unavailable = 1
+					Expect(podS.format()).To(ContainSubstring(fmt.Sprintf(" Unavailable: " + red + "1/2" + reset)))
+				})
+			})
+		})
+	})
+
+	Describe("podChecker", func() {
+		var (
+			podC          podChecker
+			ctx           = context.Background()
+			namespace     = "namespaceTest"
+			deploymentApp = "deploymentTest"
+			deployments   = []string{deploymentApp}
+			daemonSetApp  = "daemonSetTest"
+			daemonSets    = []string{daemonSetApp}
+
+			depLabels = map[string]string{
+				"app": deploymentApp,
+			}
+
+			dsLabels = map[string]string{
+				"app": daemonSetApp,
+			}
+
+			pod = &v1.Pod{
```

Comment:
Nit. Variables should be initialized in BeforeEach blocks, to avoid interference between tests is one modifies them (either intentionally or accidentally).